python/tk_desktop/desktop_window.py

```diff
@@ -681,26 +696,24 @@ def __launch_app_proxy_for_project(self, project, pipeline_configuration_id=None
             self.__set_project_just_accessed(project)
             QtGui.QApplication.instance().processEvents()
 
-            if sys.platform == "darwin":
-                path_field = "mac_path"
-            elif sys.platform == "win32":
-                path_field = "windows_path"
-            elif sys.platform.startswith("linux"):
-                path_field = "linux_path"
-            else:
-                raise SystemError("Unsupported platform: %s" % sys.platform)
-
-            filters = [
-                ["project", "is", project],
-            ]
-
-            fields = [path_field, "users", "code"]
+            # Phase 2: Get information about the pipeline configuration.
 
+            # First get the legacy pipeline configurations, i.e. those using the *_path fields.
```

Comment:
would be great to explain this a bit more - i think it will help future readers of this code. Going back to the diagrams, would be cool if we could communicate which cases are handled by bootstrap and which ones need to be handled by desktop for legacy purposes:

![image](https://cloud.githubusercontent.com/assets/337710/22060901/a4b42ef2-dd6a-11e6-9134-826ea174a431.png)

I don't think we should call these configs 'legacy' just yet. They are basically configs that are installed via tank setup_project or the install wizard - features that we haven't deprecated (and won't deprecate any time soon! - lots of clients depend on these workflows). So maybe call it the classic or installed workflow instead?